packages/mobx-state-tree/__tests__/core/model.test.ts

```diff
@@ -17,152 +18,405 @@ test("it should call preProcessSnapshot with the correct argument", () => {
-            )
+        describe('and that type is "types.array"', () => {
+            test("it sets the default value to an empty array", () => {
+                const Model = types.model({
+                    foo: types.array(types.string)
+                })
 
-            const modelSnapshot = getSnapshot(Model.create())
-            expect(modelSnapshot).toEqual({
-                prop1: "prop1",
-                prop2: 2
+                const modelSnapshot = getSnapshot(Model.create())
+                expect(modelSnapshot).toEqual({
+                    foo: []
+                })
+            })
+        })
+        describe("and that type is another model", () => {
+            test("it sets the default value to the default of that model", () => {
+                const Todo = types.model({
+                    task: types.optional(types.string, "test")
+                })
+
+                const TodoStore = types.model("TodoStore", {
+                    todo1: types.optional(Todo, () => Todo.create())
+                })
+
+                const modelSnapshot = getSnapshot(TodoStore.create())
+                expect(modelSnapshot).toEqual({
+                    todo1: {
+                        task: "test"
+                    }
+                })
             })
         })
     })
+    describe("when a user defines a property using a function", () => {
+        // if (process.env.NODE_ENV === "production") {
+        //     test("it sets the default value to undefined in production", () => {
+        //         // @ts-ignore
+        //         const Model = types.model({
+        //             foo: () => "bar"
+        //         })
+
+        //         const modelSnapshot = getSnapshot(Model.create())
+        //         expect(modelSnapshot).toEqual({
+        //             foo: undefined
+        //         })
+        //     })
+        //     test("it does not throw an error in production", () => {
+        //         expect(() => {
+        //             // @ts-ignore
+        //             types.model({
+        //                 foo: () => "bar"
+        //             })
+        //         }).not.toThrow()
+        //     })
+        // }
+        if (process.env.NODE_ENV !== "production") {
+            test("it throws an error when not in production", () => {
+                expect(() => {
+                    // @ts-ignore
+                    types.model({
+                        foo: () => "bar"
+                    })
+                }).toThrowErrorMatchingInlineSnapshot(
+                    `"[mobx-state-tree] Invalid type definition for property 'foo', it looks like you passed a function. Did you forget to invoke it, or did you intend to declare a view / action?"`
+                )
+            })
+        }
+    })
+    describe("when a user defins a property using a plain JavaScript object", () => {
+        // test("it sets the default value to undefined in production", () => {
```

Comment:
[Basically same questions as my prior thread, but about objects instead of functions](https://github.com/mobxjs/mobx-state-tree/pull/2051/files#r1272988275) - @jamonholmgren

Reply:
In the interest of getting this merged to add more test coverage [I removed the failing specs](https://github.com/mobxjs/mobx-state-tree/pull/2051/commits/9dff9372b4aa208734ffee1c9f623c596350d753). I don't think this is a major pitfall, just a nuance I was interested in. Perhaps we'll revisit!